lib/psych/visitors/yaml_tree.rb

```diff
@@ -310,19 +310,20 @@ def visit_String o
           style = Nodes::Scalar::LITERAL
           plain = false
           quote = false
-        elsif o =~ /\n/
-          style = Nodes::Scalar::LITERAL
         elsif o == '<<'
           style = Nodes::Scalar::SINGLE_QUOTED
           tag   = 'tag:yaml.org,2002:str'
           plain = false
           quote = false
+        elsif o =~ /\n[^\Z]/  # match \n except blank line at the end of string
+          style = Nodes::Scalar::LITERAL
+        elsif o.length > line_width
```

Comment:
Oh, why we’re doing this? We can simply assign `@options[:line_width]` to an instance variable during initialization! Then it’ll be just:

```ruby
...
elsif @line_width && o.length > @line_width
...
```

You can’t beat this!
```
        false:preset      7.491M (± 3.8%) i/s -     37.440M
         true:preset      6.295M (± 5.6%) i/s -     31.384M
```

And cleaner code wins again. :smile_cat: